ui/src/dataLoaders/components/TelegrafEditorSidebar.tsx

```diff
@@ -0,0 +1,192 @@
+    plugins,
+    filter,
+  }
+}
+
+const ActivePluginList = connect<ActivePluginStateProps, {}>(
+  mstp_1,
+  null
+)(PluginList)
+
+const mstp_2 = (state: AppState): PluginStateProps => {
+  const plugins = state.telegrafEditorPlugins || []
+  const filter = state.telegrafEditor.filter
+
+  return {
+    plugins,
+    filter,
+  }
+}
+
+const AllPluginList = connect<PluginStateProps, {}>(
+  mstp_2,
+  null
+)(PluginList)
+
+interface StateProps {
+  buckets: Bucket[]
+  bucket: Bucket
+  filter: string
+  mode: 'adding' | 'indexing'
+}
+
+interface DispatchProps {
+  onSetFilter: typeof setFilter
+  onSetBucket: typeof setBucket
+  onSetMode: typeof setMode
+}
+
+type ListPlugin = TelegrafEditorBasicPlugin | TelegrafEditorBundlePlugin
+interface OwnProps {
+  onJump: (which: TelegrafEditorActivePlugin) => void
+  onAdd: (which: ListPlugin) => void
+}
+
+type TelegrafEditorSidebarProps = StateProps & DispatchProps & OwnProps
+
+class TelegrafEditorSideBar extends PureComponent<TelegrafEditorSidebarProps> {
+  render() {
+    const {bucket, buckets, filter, mode} = this.props
+    return (
+      <Grid.Column widthXS={Columns.Three} style={{height: '100%'}}>
+        <BucketDropdown
+          buckets={buckets}
+          selectedBucketID={bucket.id}
+          onSelectBucket={this.handleSelectBucket}
+        />
+        <Input
+          className="wizard-step--filter"
+          size={ComponentSize.Small}
+          icon={IconFont.Search}
+          value={filter}
+          onBlur={this.handleFilterBlur}
+          onChange={this.handleFilterChange}
+          placeholder="Filter Plugins..."
+        />
+        <Tabs.Container
+          orientation={Orientation.Horizontal}
+          style={{height: 'calc(100% - 96px)', marginTop: '18px'}}
+        >
+          <Tabs>
+            <Tabs.Tab
+              id="lookup"
+              text="Plugin Lookup"
+              active={mode === 'indexing'}
+              onClick={() => {
+                this.handleSetMode('indexing')
+              }}
+            />
+            <Tabs.Tab
+              id="add"
+              text="Add Plugins"
+              active={mode === 'adding'}
+              onClick={() => {
+                this.handleSetMode('adding')
+              }}
+            />
+          </Tabs>
+          <Tabs.TabContents padding={ComponentSize.Small}>
+            {mode === 'indexing' && (
+              <ActivePluginList onClick={this.handleJump} />
+            )}
+            {mode === 'adding' && <AllPluginList onClick={this.handleAdd} />}
+          </Tabs.TabContents>
+        </Tabs.Container>
+      </Grid.Column>
+    )
+  }
+
+  private handleJump = which => {
+    this.props.onJump(which)
+  }
+
+  private handleAdd = which => {
+    this.props.onAdd(which)
+  }
+
+  private handleSetMode = state => {
+    this.props.onSetMode(state)
+  }
+
+  private handleSelectBucket = evt => {
+    this.props.onSetBucket(evt)
+  }
+
+  private handleFilterBlur = evt => {
```

Comment:
why do you wrap these in a functions instead of just passing in this.props.handleFilterBlur on line 104? and for all the rest?

Reply:
oh.. they used to do something else for another thing and i forgot about them. will clean up